src/views/portfolio/projects/ProjectList.vue

```diff
@@ -168,13 +168,35 @@
           dataOff: '\u2715'
         },
         columns: [
-          {
+        {
             title: this.$t('message.project_name'),
             field: "name",
             sortable: true,
             formatter(value, row, index) {
-              let url = xssFilters.uriInUnQuotedAttr("../projects/" + row.uuid);
-              return `<a href="${url}">${xssFilters.inHTMLData(value)}</a>`;
+              let name_url = xssFilters.uriInUnQuotedAttr("../projects/" + row.uuid);
+              let projectName = `<a href="${name_url}">${xssFilters.inHTMLData(value)}</a>`;
+              return projectName
+            }
+          },
+          {
+            title: this.$t('message.tags'),
+            field: "tags",
+            sortable: true,
+            visible: false,
+            formatter(value, row, index) {
+              if (!row.tags) {
+                return "";
+              } else {
+                let tagNames = ""
+                  for (let i=0; i<row.tags.length; i++) {
+                  let tag = row.tags[i].name
+                  let tag_url = xssFilters.uriInUnQuotedAttr("../projects/?tag=" + tag);
+                  let tagName = `<a href="${tag_url}" target="_self" class="">${tag}</a>`
+                  tagNames += tagName
+                  if (i < row.tags.length-1) tagNames += ", "
+                }
+                return tagNames
```

Comment:
There are several comments here.


1. the indentations are different or not correct (line 191).
2. missing semicolons
3. if statement without curly braces
4. used `let` where `const` would be possible
5. **a possible persistent XSS vulnerability in line 194 because `tag` is not escaped there!**
6. the tag-value in the url is not url-encoded. For example the tag "=foo" leads to an invalid URL.

I would recommend this.
```js
return row.tags?.map(tag => `<a href="../projects/?tag=${xssFilters.uriComponentInUnQuotedAttr(tag)}">${xssFilters.inHTMLData(tag)}</a>`).join(', ') || '';
```

Reply:
I think in a converstaion somewhere we concluded escaping should be done automatically here by bootstrap, but you're right, we need to do it ourselves. For the rest I am sometimes tempted to just copy and paste some code from other places and leave it in "imperative" style as it's easier to read.

but I've applied your oneline now.